src/test_typing_extensions.py

```diff
@@ -7165,13 +7165,58 @@ def test_attributes(self):
         self.assertEqual(ListOrSetT.__type_params__, (T,))
         self.assertEqual(ListOrSetT.__parameters__, (T,))
 
+        subscripted = ListOrSetT[int]
```

Comment:
To some extent this issue predated your PR, but this PR makes it worse, so I think this would be a good opportunity to fix it: this `test_attributes()` method is huge! I think it would be worthwhile to split it into three methods:
- `test_attributes`
- `test_substitution_with_alias_generic_over_typevar`
- `test_substitution_with_alias_generic_over_typevartuple`

This would also clarify that for everything except the first few assertions in this test, we're really testing the substitution behaviour rather than testing which attributes exist on the `TypeAliasType` instances.